ManageRDSessions/ManageRDSessions.psm1

```diff
@@ -27,10 +38,29 @@ function Get-sbRDSession {
             Connected    = "STATE_CONNECTED"
             Any          = "*"
         }
-        Write-Verbose "Querying RD Session Collection for [$SessionState] sessions"
-        $sessions = Get-RDUserSession | Where-Object {
-            $_.SessionState -like $stateLookup.$SessionState -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleMins
-        }
+
+        if ($UserName) {
+            Write-Verbose "Querying RD Session Collection for users like [$UserName]"
+            $sessions = Get-RDUserSession | Where-Object {
+                $_.UserName -like "*$UserName*"
+            }
+        } else {
+            if ($IncludeSelf) {
+                Write-Verbose "Querying RD Session Collection for [$SessionState] sessions - including [$env:USERNAME]"
+                $sessions = Get-RDUserSession | Where-Object {
+                    $_.SessionState -like $stateLookup.$SessionState`
+                        -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleMins`
+                }
+            } else {
+                Write-Verbose "Querying RD Session Collection for [$SessionState] sessions"
+                $sessions = Get-RDUserSession | Where-Object {
+                    $_.SessionState -like $stateLookup.$SessionState`
+                        -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleMins`
+                        -and $_.UserName -notlike "*$env:USERNAME*"
```

Comment:
This might be too vague - let me know if you think just a `-not` is more appropriate.